filebeat/beater/filebeat.go

```diff
@@ -81,11 +81,8 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 		return err
 	}
 
-	// Start up spooler
 	fb.spooler.Start()
-
-	// registrar records last acknowledged positions in all files.
-	go fb.registrar.Run()
+	fb.registrar.Start()
```

Comment:
with spooler forwarding to registrar + registrar having to shutdown after spooler, I'd put registrar.Start before spooler. Keep a stack-like order of process startup/shutdown.